src/com/hms_networks/americas/sc/realtimedata/RealTimeDataQueueManager.java

```diff
@@ -0,0 +1,316 @@
+   * iterators to look at the next tag in the list of group A tags, and remove the retrieved tag
+   * from the list of tags.
+   *
+   * <p>If there are no DataPoints left, null will be returned.
+   *
+   * @return returns a DataPoint object
+   */
+  public static DataPoint getGroupANextData() {
+    DataPoint data = null;
+    RealTimeTagDataPointManager dataPoints =
+        ((RealTimeTagDataPointManager) tagManagerA.get(tagManagerIteratorA));
+    if (dataPoints.getSize() > 0) {
+      data = dataPoints.removeDataPoint();
+      tagManagerIteratorA += 1;
+      tagManagerIteratorA = (tagManagerIteratorA % tagsGroupA.size());
+    }
+    return data;
+  }
+
+  /**
+   * getGroupBNextData will group a single data point object from the tags in group B, rotate
+   * iterators to look at the next tag in the list of group B tags, and remove the retrieved tag
+   * from the list of tags.
+   *
+   * <p>If there are no DataPoints left, null will be returned.
+   *
+   * @return returns a DataPoint object
+   */
+  public static DataPoint getGroupBNextData() {
+    DataPoint data = null;
+    RealTimeTagDataPointManager dataPoints =
+        ((RealTimeTagDataPointManager) tagManagerB.get(tagManagerIteratorB));
+    if (dataPoints.getSize() > 0) {
+      data = dataPoints.removeDataPoint();
+      tagManagerIteratorB += 1;
+      tagManagerIteratorB = (tagManagerIteratorB % tagsGroupB.size());
+    }
+    return data;
+  }
+
+  /**
+   * getGroupCNextData will group a single data point object from the tags in group C, rotate
+   * iterators to look at the next tag in the list of group C tags, and remove the retrieved tag
+   * from the list of tags.
+   *
+   * <p>If there are no DataPoints left, null will be returned.
+   *
+   * @return returns a DataPoint object
+   */
+  public static DataPoint getGroupCNextData() {
+    DataPoint data = null;
+    RealTimeTagDataPointManager dataPoints =
+        ((RealTimeTagDataPointManager) tagManagerC.get(tagManagerIteratorC));
+    if (dataPoints.getSize() > 0) {
+      data = dataPoints.removeDataPoint();
+      tagManagerIteratorC += 1;
+      tagManagerIteratorC = (tagManagerIteratorC % tagsGroupC.size());
+    }
+    return data;
+  }
+
+  /**
+   * getGroupDNextData will group a single data point object from the tags in group D, rotate
+   * iterators to look at the next tag in the list of group D tags, and remove the retrieved tag
+   * from the list of tags.
+   *
+   * <p>If there are no DataPoints left, null will be returned.
+   *
+   * @return returns a DataPoint object
+   */
+  public static DataPoint getGroupDNextData() {
+    DataPoint data = null;
+    RealTimeTagDataPointManager dataPoints =
+        ((RealTimeTagDataPointManager) tagManagerD.get(tagManagerIteratorD));
+    if (dataPoints.getSize() > 0) {
+      data = dataPoints.removeDataPoint();
+      tagManagerIteratorD += 1;
+      tagManagerIteratorD = (tagManagerIteratorD % tagsGroupD.size());
+    }
+    return data;
+  }

```

Comment:
Too much  repeating code. Make one method that takes the tag group as a parameter.